config/init.go

```diff
@@ -0,0 +1,83 @@
+package config
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"io"
+
+	"github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+func Init(out io.Writer) (*Config, error) {
+	identity, err := CreateIdentity(out)
+	if err != nil {
+		return nil, err
+	}
+
+	return InitWithIdentity(identity)
+}
+
+func InitWithIdentity(identity Identity) (*Config, error) {
+	conf := &Config{
+		// setup the node's default addresses.
+		Addresses: Addresses{
+			Admin:     "/ip4/0.0.0.0/tcp/3001",
+			Discovery: "/ip4/0.0.0.0/tcp/3002",
+			Ingest:    "/ip4/0.0.0.0/tcp/3003",
+			Finder:    "/ip4/0.0.0.0/tcp/3000",
+		},
+
+		Datastore: Datastore{
```

Comment:
I think I don't understand the datastore config (I may be missing something). Are we planning to support different types and according to the type a different datastore is initialized?

Reply:
I was thinking that we would support different datastores, but really there is no need to.  I should probably remove his part of the config.